### PR TITLE
Reexport components already migrated to PF

### PR DESCRIPTION
**Pull request by fhlavac**
[RHCLOUD-26558](https://issues.redhat.com/browse/RHCLOUD-26558)

Reexported components already migrated to PF. In some cases added Red Hat specific stuff to not cause issues for the existing imports.

### CODE REVIEW

**Comment by fhlavac** [comment]
Added eslint rule to warn about using deprecated imports
image

File: packages/eslint-config/lib/rules/disallow-pf-migrated-components.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    },
+    fixable: 'code',
+    schema: [],
+  },
+  create: function (context) {
+    return {
+      ImportDeclaration(node) {
+        const sourceValue = node.source.value;
+        const specifiers = node.specifiers;
+        const forbiddenImports = ['ErrorBoundary', 'NotAuthorized', 'ErrorState'];
```
Comment (karelhala) [inline packages/eslint-config/lib/rules/disallow-pf-migrated-components.js:22]: Can we move this out of the function so it's more visible and easier for us to add new deprecated values?

**Review by karelhala — COMMENTED** [review]
Looking good! Just a minor question.

File: packages/components/src/ErrorBoundary/ErrorBoundary.tsx [hunk 2]
```diff
@@ -1,75 +1,13 @@
...
-      );
-    }
-
-    return this.props.children;
-  }
-}
-
-export default ErrorBoundaryPage;
+import React from 'react';
+import { ErrorBoundary as ErrorBoundaryPF } from '@patternfly/react-component-groups';
```
Comment (Hyperkid123) [inline packages/components/src/ErrorBoundary/ErrorBoundary.tsx:2]: We should be exporting the `dist/dynamic` absolute path to preserve the module federation compatibility.

**Review by Hyperkid123 — CHANGES_REQUESTED** [review]

**Review by Hyperkid123 — APPROVED** [review]

**Comment by Hyperkid123** [comment]
@fhlavac the groups repo will be ported to pf4@5 soon. Not sure if we want to wait a little bit.

**Review by karelhala — APPROVED** [review]

**Review by Hyperkid123 — APPROVED** [review]

**Pull request merged**